client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/RestService.java

```diff
@@ -127,6 +129,7 @@
 
   private UrlList baseUrls;
   private SSLSocketFactory sslSocketFactory;
+  private Optional<HostnameVerifier> hostnameVerifier = Optional.empty();
```

Comment:
Storing an `Optional` in a field is usually an anti-pattern, can just use a nullable `HostnameVerifier` here